### ISSUE TITLE
Two big concerns about “Error messages collocated”

### ISSUE THREAD

**Issue by julierawe** [w3c/wcag3#790]
**Issue description:**
**First big concern:** The [WCAG 3 definition of “visually collocated”](https://www.w3.org/TR/wcag-3.0/#dfn-visually-collocated-with) is hard to understand: “in reasonable proximity within the viewport, regardless of magnification or zoom level, to sufficiently associate the error message to the field in error.” 

We also think it’s hard to understand the second half of the outcome sentence: “or the focus is moved to the error message and a mechanism is available to move to the input that is in error.”

We found this term and provision hard to understand. A simpler definition would help, such as “place side by side or in a particular relation.” 

**Second big concern:** Will this provision make the content usable by COGA user groups? We know this focuses on screen reader users like JAWS, which is fine, but there should be support for COGA use groups for the same problems. It could be in a separate provision or included here.

The scenario: People with cognitive and learning disabilities often do not notice the error message or that anything has changed from when they pressed the submit button (they may be looking at the submit button). We may press “submit” and think that nothing happened. So we use the mouse to press submit a few more times and then assume it does not work. We may not have noticed that the focus was moved as we are using a mouse. We may be locked out of our account or just give up, assuming the site does not work. 

COGA user groups need three things:

- We need to see and notice the error message. Even without being a good reader, etc. For example, a - banner in the expected location, where we are looking. 
- And we need to see the source of the problem (icons, color etc) (see https://github.com/w3c/wcag3/issues/787)
- We need to understand what to do about it. (see https://github.com/w3c/wcag3/issues/788)

Here is why we feel it is not giving equitable support in this provision:
**1st option in the outcome sentence: “visually collocated with the error source.”** Typically the user is looking at the submit button. They will need to notice the error message and sometimes may need to scroll to find the error message (especially if they are low vision).

**Second option in the outcome sentence: “or the focus is moved to the error message and a mechanism … “** This option helps visually impaired users who are using screen readers. If you allow a change of focus that is far from the source of the error, sighted users may not notice the change in focus.

What would help is if: 

- the messages are in a very prominent and expected location, like a clear list of errors in a banner, so that everyone notices them, AND 
- at the source of an error, have a color change and icons, so they are easy to find. (See https://github.com/w3c/wcag3/issues/787 on multiple ways) 

However, as it stands, COGA users will not be helped because:

- If the focus moves the user might not see it or hear it, without a traditional screen reader. 
- Coga users typically can not use a complex screen reader like jaws. If they are using text to speech, it might be an extension or simple text to speech.  aria -describedby will not do anything to help COGA user groups, but aria-describedby is a valid mechanism here.

We need an error message very noticeable such as

- at the top and /or 
- by the submit button 
- In both cases, we would still need it to use multiple ways to make it stand out (see Multiple ways provision)

Thanks,
The COGA Task Force
